server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/schema/NormalSchemaFetcher.java

```diff
@@ -122,6 +180,19 @@ void processNormalTimeSeries(
                 .collect(Collectors.toList()),
             indexOfDevicesWithMissingMeasurements,
             indexOfMissingMeasurementsList);
+    if (hasUnFetchedLogicalView) {
+      List<String> fullPathsNeedRefetch = new ArrayList<>();
+      for (Pair<List<Integer>, List<String>> pair : missedIndexAndPathStringOfViewList) {
+        fullPathsNeedRefetch.addAll(pair.right);
+      }
+      reFetchFullPathsAndMerge(remoteSchemaTree, fullPathsNeedRefetch);
+    }
```

Comment:
Merge these two fetch operations to make sure only fetch once.

Reply:
fetch optimism：
fetch 1: always fetch with a pattern tree with 1 fetching if there is view.
fetch 2: always fetch with full paths if there is view.